src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/CheckUtil.java

```diff
@@ -349,4 +351,16 @@ public static String getJavadocTokenText(int[] tokens, int... subtractions) {
         return result.toString();
     }
 
+    public static String getLineSeparatorForFile(String filepath, Charset charset)
+            throws IOException {
+        final String result;
+        if (new FileText(new File(filepath), charset.name())
+                .getFullText().chars().anyMatch(character -> character == '\r')) {
```

Comment:
do we need to read whole file text ?
can we read only first line https://stackoverflow.com/questions/24666805/java-only-read-first-line-of-a-file/46239829 ? or smth similar to https://stackoverflow.com/a/45168182/1015848 ?

Reply:
Those approaches can of course not be used.
They read the first line until the line separator but do not include the line separator.
But I can of course easily stop at an `\n`.
The method is heuristic anyway.
Like I have it now, if you have 1000 LF in a file and one CRLF, it will be recognized as CRLF.
When I stop at the first LF, then it only works properly if the first line separator is CRLF, but that might be ok for the use-case.

Reply:
changed to that

Reply:
ok, it is test code, it can inefficient a bit.

Reply:
Do you think it is still inefficient? o_O